models/progresses.js

```diff
@@ -47,9 +48,40 @@ const createProgressDocument = async (progressData) => {
  * @throws {Error} If the userId or taskId is invalid or does not exist.
  **/
 const getProgressDocument = async (queryParams) => {
+  const { dev } = queryParams;
   await assertUserOrTaskExists(queryParams);
   const query = buildQueryToFetchDocs(queryParams);
   const progressDocs = await getProgressDocs(query);
+
+  if (dev) {
+    try {
+      const uniqueUserIds = [...new Set(progressDocs.map((doc) => doc.userId))];
+      const batchSize = 500;
+
+      const batches = Array.from({ length: Math.ceil(uniqueUserIds.length / batchSize) }, (_, index) =>
+        uniqueUserIds.slice(index * batchSize, index * batchSize + batchSize)
+      );
+
+      const batchPromises = batches.map((batch) => Promise.all(batch.map((userId) => fetchUser({ userId }))));
```

Comment:
Please refactor this and why we need this?

Reply:
Achintya asked me to use the batches approach as Promise.all could cause performance issues for large datasets. You can check the prevoius resolved messages

Reply:
resolve